### PR TITLE
Improvements to MetaZoneCalculator

### PR DESCRIPTION
**Pull request by sffc**
Follow-up to #2122

### CODE REVIEW

File: components/datetime/src/metazone.rs [hunk 1]
```diff
@@ -28,22 +27,19 @@ impl MetaZoneCalculator {
...
-        L: Into<Locale>,
-        ZP: DataProvider<MetaZonePeriodV1Marker> + ?Sized,
+        P: DataProvider<MetaZonePeriodV1Marker> + ?Sized,
     {
-        let locale = locale.into();
         let metazone_period = zone_provider
             .load(DataRequest {
-                locale: &DataLocale::from(locale),
+                locale: Default::default(),
                 metadata: Default::default(),
```
Comment (samchen61661) [inline components/datetime/src/metazone.rs:43]: thanks for fixing it!

File: components/datetime/src/metazone.rs [hunk 2]
```diff
@@ -28,22 +27,19 @@ impl MetaZoneCalculator {
...
-    /// let mzc = MetaZoneCalculator::new(
-    ///     locale!("en"),
-    ///     &provider,
-    /// );
+    /// let mzc = MetaZoneCalculator::try_new(&provider);
     ///
     /// assert!(mzc.is_ok());
     /// ```
-    pub fn new<L, ZP>(locale: L, zone_provider: &ZP) -> Result<Self, DateTimeFormatterError>
+    pub fn try_new<P>(
```
Comment (samchen61661) [inline components/datetime/src/metazone.rs]: You might forget to update the doc on line 14.

**Comment by sffc** [comment]
@samchen61661 You should have permissions to approve the PR. Could you do that? Then we can merge.

**Review by samchen61661 — APPROVED** [review]

**Comment by samchen61661** [comment]
> @samchen61661 You should have permissions to approve the PR. Could you do that? Then we can merge.

Sure!

File: components/datetime/src/metazone.rs (hunk 2 above)
Comment (sffc) [reply components/datetime/src/metazone.rs]: Yep; fixed in https://github.com/unicode-org/icu4x/pull/2274/commits/8d2ea5e9ccc1936846b9797d32bcdb88c0a01df5

**Pull request merged**